web/packages/teleterm/src/ui/Vnet/VnetSliderStep.tsx

```diff
@@ -129,3 +103,74 @@ const ErrorText = (props: PropsWithChildren) => (
     {props.children}
   </Text>
 );
+
+/**
+ * DnsZones displays the list of currently proxied DNS zones, as understood by the VNet admin
+ * process. The list is cached in the context and updated when the VNet panel gets opened.
+ *
+ * As for 95% of users the list will never change during the lifespan of VNet, the VNet panel always
+ * optimistically displays previously fetched results while fetching new list.
+ */
+const DnsZones = () => {
+  const { listDNSZones, listDNSZonesAttempt } = useVnetContext();
+  const dnsZonesRefreshRequestedRef = useRef(false);
+
+  useEffect(function refreshListOnOpen() {
+    if (!dnsZonesRefreshRequestedRef.current) {
+      dnsZonesRefreshRequestedRef.current = true;
+      listDNSZones();
+    }
+  }, []);
+
+  if (listDNSZonesAttempt.status === 'error') {
+    return (
+      <Text p={textSpacing}>
+        <ConnectionStatusIndicator status="warning" inline mr={2} />
+        VNet is working, but Teleport Connect could not fetch DNS zones:{' '}
+        {listDNSZonesAttempt.statusText}
+        <ButtonSecondary
+          ml={2}
+          size="small"
+          type="button"
+          onClick={listDNSZones}
+        >
+          Retry
+        </ButtonSecondary>
+      </Text>
+    );
+  }
+
+  if (
+    listDNSZonesAttempt.status === '' ||
+    (listDNSZonesAttempt.status === 'processing' && !listDNSZonesAttempt.data)
+  ) {
+    return (
+      <Text p={textSpacing}>
+        <ConnectionStatusIndicator status="processing" inline mr={2} />
+        Updating list of DNS zones.
```

Comment:
What do you think about adding an ellipsis to this message?
I think it signals a pending action better than a dot:
<img width="494" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/38f2034d-66d2-407a-84f1-1728c445c900">